### PR TITLE
This makes the lint script use multiprocessing to speed it up.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

I got tired of waiting for it to run, so I added some of the "worker" queue code that I wrote for the assets-for-api-docs generator.

I also tried out putting all the files in one call to clang-tidy with the `-p` argument, but that was still a lot slower because it runs them all on one core. This runs separate jobs for each file, simultaneously, and then reports the results at the end (associated with each file, of course).

### CODE REVIEW

**Comment by gaaclarke** [comment]
I'll have to give this a bit closer read tomorrow.  At first blush I think you should pull the work you did here into a package for pub.  I did something similar here: https://pub.dev/packages/pmap   It spawns isolates to perform tasks, we should have something similar for subprocesses like you've done.

File: ci/lint.dart [hunk 1]
```diff
@@ -5,17 +5,279 @@
...
+          completer.complete(completedJobs);
+        }
+      }
+      jobDone.complete(output);
+      _printReport();
+    }
+    return jobDone.future;
+  }
+
+  Future<Map<WorkerJob, ProcessRunnerResult>> startWorkers(
```
Comment (gaaclarke) [inline ci/lint.dart:259]: nit: This would be more efficient and robust if it returned `Stream<Tuple<WorkerJob, ProcessRunnerResult>>`.

Comment (gspencergoog) [reply ci/lint.dart:259]: Cool, thanks for the suggestion, I'll do that.

**Comment by gspencergoog** [comment]
Yeah, I agree: I've already started the process of releasing it in a separate `google/` github repo, since it's not Flutter specific, and I don't want to put it into the `packages` repo.

**Comment by gspencergoog** [comment]
Actually, there's one consideration: I modified this to not use `package/process.dart`, since this script doesn't have a `pubspec.yaml` to add the dependency to, but I actually think that the published version is more generally useful if it does: you can mock it for tests, etc.

Should I make this into a "real" dart app with a pubspec.yaml?

**Comment by gaaclarke** [comment]
> Should I make this into a "real" dart app with a pubspec.yaml?

Yea, sounds like that makes the most sense.  Probably would have happened eventually anyways.

**Comment by gspencergoog** [comment]
OK, I'm in the process of releasing the process_runner package (have to get some approvals).  Once that is done, I'll update this with that dependency and we can move forward.  Until then, I'll switch this back to a "draft" PR.

**Comment by gspencergoog** [comment]
OK, the package is released: https://pub.dev/packages/process_runner

I'll incorporate it into the script.

File: ci/bin/lint.dart [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  parser.addOption('checks',
+      help: 'Perform the given checks on the code. Defaults to the empty '
+          'string, indicating all checks should be performed.',
+      defaultsTo: '');
+  final ArgResults options = parser.parse(arguments);
+
+  verbose = options['verbose'] as bool;
+
+  if (options['help'] as bool) {
+    _usage(parser);
```
Comment (gaaclarke) [inline ci/bin/lint.dart:154]: Shouldn't execution happen after _usage?

**Comment by gaaclarke** [comment]
I'd use `led` to fire off a few builds before landing this to make sure it doesn't break CI.

File: ci/bin/lint.dart (hunk 2 above)
Comment (gspencergoog) [reply ci/bin/lint.dart:164]: Yes, and it does: `_usage` exits.

**Pull request merged**